### ISSUE TITLE
Add a cli command to get environment info

### ISSUE THREAD

**Issue by freddyaboulton** [gradio-app/gradio#4598]
- [ ] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
It might make it easier for people to file bug reports if we shipped an easy command to list the OS, gradio version, and versions of gradio/gradio-client dependencies.

**Describe the solution you'd like**  
Something like `gradio environment`

We rarely have issues due to third party deps. So listing the gradio version and os may be enough.

**Additional context**

**Comment by pngwn** [comment]
This would be cool. In the JS community `npx envinfo` is used pretty extensively for this purpose and it is very useful.

**Comment by abidlabs** [comment]
Hi @varshneydevansh I think a good starting point would be `gradio/cli.py` which routes various CLI commands to appropriate python functions. For example, if someone types "gradio deploy ...", it calls `gradio.deploy_space.deploy()`

**Comment by varshneydevansh** [comment]
Hi @abidlabs how do I find the info about the dependencies? I did try to look in the codebase but found myself a little lost.

Currently, I added this `print_environment_info()` in the `cli.py` but will move it into another new file.

```python
import sys
import platform
import gradio.deploy_space
import gradio.reload

def cli():
    args = sys.argv[1:]
    if len(args) == 0:
        raise ValueError("No file specified.")
    elif args[0] == "deploy":
        gradio.deploy_space.deploy()
    elif args[0] == "environment":
        print_environment_info()
    else:
        gradio.reload.main()

def print_environment_info():
    print("Gradio Environment Information:")
    print(f"Operating System: {platform.system()}")
    print(f"Gradio Version: {gradio.__version__}")
    print("Gradio Dependencies:")
    for package in gradio.__dependencies__:
        print(f"{package}: {gradio.__dependencies__[package]}")

```

**Comment by varshneydevansh** [comment]
I think I did found https://github.com/gradio-app/gradio/tree/main/client/python

**Comment by abidlabs** [comment]
Fixed in #4915, thanks @varshneydevansh!
